src/hotspot/share/oops/instanceKlass.cpp

```diff
@@ -3829,6 +3832,59 @@ void InstanceKlass::print_class_load_logging(ClassLoaderData* loader_data,
   }
 }
 
+void InstanceKlass::print_class_load_cause_logging() const {
+  bool log_cause_native = log_is_enabled(Info, class, load, cause, native);
+  if (log_cause_native || log_is_enabled(Info, class, load, cause)) {
+    JavaThread* current = JavaThread::current();
+    ResourceMark rm;
+    const char* name = external_name();
+
+    if (LogClassLoadingCauseFor != nullptr &&
+        strstr(name, LogClassLoadingCauseFor) == nullptr) {
+        return;
+    }
+
+    // Log Java stack first
+    {
+      LogMessage(class, load, cause) msg;
+      NonInterleavingLogStream info_stream{LogLevelType::Info, msg};
+
+      info_stream.print_cr("Java stack when loading %s:", name);
+      current->print_stack_on(&info_stream);
+    }
+
+    // Log native stack second
+    if (log_cause_native) {
+      // Log to string first so that lines can be indented
+      stringStream stack_stream;
+      char buf[O_BUFLEN];
+      frame f = os::current_frame();
+      VMError::print_native_stack(&stack_stream, f, current, true /*print_source_info */,
+                                  -1 /* max stack_stream */, buf, O_BUFLEN);
+
+      LogMessage(class, load, cause, native) msg;
+      NonInterleavingLogStream info_stream{LogLevelType::Info, msg};
+      info_stream.print_cr("Native stack when loading %s:", name);
+
+      // Print each native stack line to the log
```

Comment:
`LogStream`  extends `OutputStream` which has some support for indentation. I thought there should be a way to simplify this, but given you have to break into lines anyway the `\t` seems easier than using any of the built-in indentation functionality.